core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -29,10 +29,15 @@ final case class Uri(
   fragment: Option[Fragment] = None)
   extends QueryOps with Renderable
 {
+  import Uri._
+
   def withPath(path: Path): Uri = copy(path = path)
 
-  def /(newFragment: Path): Uri = {
-    val newPath = if (path.isEmpty || path.last != '/') path + "/" + newFragment else path + newFragment
+  def /(newSegment: Path)(implicit charset: Charset = Charset.`UTF-8`): Uri = {
```

Comment:
It turns out we're already heavily biased toward UTF-8: the query and fragment rendering of the Uri assume this.